tensorflow/core/common_runtime/dml/dml_gpu_event.h

```diff
@@ -77,8 +77,23 @@ struct DmlGpuEvent {
     WaitForSingleObject(h.get(), INFINITE);
 #else
     // Nullptr event blocks CPU until completion (creates and waits on an event
-    // internally)
-    DML_CHECK_SUCCEEDED(fence->SetEventOnCompletion(fence_value, nullptr));
+    // internally).  This API returns an errur on WSL if the fence value
+    // is not reached after 10 seconds (although this may be changed in future versions 
+    // of the WSL kernel). To work around this, failing calls are re-issued
+    // if they took a minimum length of time (arbitrarily, 1 second). 
+    while (true) {
```

Comment:
Seems like a max timeout before bailing entirely so would be reasonable... maybe a minute or so.

Reply:
Neither way is perfect, but as-is I think this provides closer parity with native Windows.  We can revisit and consider a broader timeout policy if needed.